sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -118,7 +118,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def option(key: String, value: String): DataFrameReader = {
-    this.extraOptions += (key -> value)
+    this.extraOptions = this.extraOptions.updated(key, value)
```

Comment:
Because we already implement `+`, can you just write `this.extraOptions = this.extraOptions.updated + (key -> value)`? I think that is how Scala 2.12 desugared this already, but 2.13 won't. If so that's the most straightforward change, doesn't need a new method.

Reply:
OK, I replaced `updated` with `+`. 

I had to override it in `2.13` version of `CaseInsensitiveMap`, because a standard `Map` would be returned from `+`.

Reply:
Weird, it isn't needed in 2.12? OK fine. It could be OK to add it to 2.12 too for completeness

Reply:
It was already in 2.12, but not in 2.13